packages/opam-monorepo/opam-monorepo.0.2.2/opam

```diff
@@ -0,0 +1,47 @@
+opam-version: "2.0"
+synopsis: "Assemble and manage fully vendored Dune repositories"
+description: """
+The opam monorepo plugin provides a convenient interface to bridge the
+opam package manager with having a local copy of all the source
+code required to build a project using the dune build tool."""
+maintainer: ["anil@recoil.org"]
+authors: [
+  "Anil Madhavapeddy" "Nathan Rebours" "Lucas Pluvinage" "Jules Aguillon"
+]
+license: "ISC"
+homepage: "https://github.com/ocamllabs/opam-monorepo"
+bug-reports: "https://github.com/ocamllabs/opam-monorepo/issues"
+depends: [
+  "dune" {>= "2.8"}
+  "ocaml" {>= "4.08.0"}
+]
+dev-repo: "git+https://github.com/ocamllabs/opam-monorepo.git"
+build: [ "dune" "build" "-p" name "-j" jobs "@install" "@runtest" {with-test} ]
+flags: [ plugin ]
+depexts: [
+  ["devel/pkgconf"] {os = "openbsd"}
+  ["pkg-config"] {os-family = "debian"}
+  ["pkg-config"] {os = "macos" & os-distribution = "homebrew"}
+  ["pkgconf"] {os = "freebsd"}
+  ["pkgconf"] {os-distribution = "alpine"}
+  ["pkgconf"] {os-distribution = "arch"}
+  ["pkgconf-pkg-config"] {os-distribution = "centos" & os-version >= "8"}
+  ["pkgconf-pkg-config"] {os-distribution = "rhel" & os-version >= "8"}
+  ["pkgconfig"] {os-distribution = "fedora"}
+  ["pkgconfig"] {os-distribution = "mageia"}
+  ["pkgconfig"] {os-distribution = "nixos"}
+  ["pkgconfig"] {os-distribution = "ol"}
+  ["pkgconfig"] {os = "macos" & os-distribution = "macports"}
+  ["pkgconfig"] {os-distribution = "centos" & os-version <= "7"}
+  ["pkgconfig"] {os-distribution = "rhel" & os-version <= "7"}
+  ["system:pkgconf"] {os = "win32" & os-distribution = "cygwinports"}
+]
```

Comment:
```suggestion
  "conf-pkg-config" {build}
]
dev-repo: "git+https://github.com/ocamllabs/opam-monorepo.git"
build: [ "dune" "build" "-p" name "-j" jobs "@install" "@runtest" {with-test} ]
flags: [ plugin ]
```
I was under the impression that this alternative way of handling depexts was more or less agreed in the discussion leading to https://github.com/ocamllabs/opam-monorepo/issues/152

Reply:
I'm fine with this diff being in the opam-repository, and also with @NathanReb's version in the upstream repository lockfile.

Reply:
Having further thought about this I'm not sure how well it fits with opam-monorepo. One of the reason to use lockfiles is to shield itself from upstream changes in opam-repository to ensure reproducibility.

I understand that in this case there are few chances that changes to `conf-*` packages actually result in breaking something but I learned to adopt a defensive posture with those things.

That being said, opam-monorepo is using itself mostly to get direct feedback on the tool and not because we really care about reproducibility so I'm happy with this change here.

Reply:
Agreed @NathanReb. Let's treat this as an experiment: start by vendoring the OCaml dependencies here with monorepo and see how it goes, and keep an eye on depexts and see if it becomes a problem.  We'll learn with every new monorepo-using release. Perhaps its worth adding a `tag` field to mark the package as a monorepo user?

Reply:
We could do that indeed! I'm also still strongly considering using a variable to control which dependencies are vendored so that the opam metadata reflect what the actual dependencies are while still not installing them by default and relying on the vendoring instead.

The presence of this variable in an opam file would act as a tag as it would easily allow us to track down which packages use opam-monorepo.

You're suggesting the use of tag for that purpose right? Since we're actively trying to make the releases compatible with default opam, I don't expect opam itself to interpret this tag.